src/perf_counters.h

```diff
@@ -124,44 +126,44 @@ class PerfCounters final {
 // Typical usage of the above primitives.
 class PerfCountersMeasurement final {
  public:
-  PerfCountersMeasurement(PerfCounters&& c)
-      : counters_(std::move(c)),
-        start_values_(counters_.IsValid() ? counters_.names().size() : 0),
-        end_values_(counters_.IsValid() ? counters_.names().size() : 0) {}
+  PerfCountersMeasurement(const std::vector<std::string>& counter_names);
+  ~PerfCountersMeasurement();
 
-  bool IsValid() const { return counters_.IsValid(); }
+  bool IsValid() const { return counters_->IsValid(); }
 
   BENCHMARK_ALWAYS_INLINE void Start() {
     assert(IsValid());
     // Tell the compiler to not move instructions above/below where we take
     // the snapshot.
     ClobberMemory();
-    counters_.Snapshot(&start_values_);
+    valid_read_ &= counters_->Snapshot(&start_values_);
     ClobberMemory();
   }
 
-  BENCHMARK_ALWAYS_INLINE std::vector<std::pair<std::string, double>>
-  StopAndGetMeasurements() {
+  BENCHMARK_ALWAYS_INLINE bool Stop(std::vector<std::pair<std::string, double>> *measurements) {
```

Comment:
you assume `measurements` is non-null, so why not receive it as a reference instead of a pointer?

Reply:
@dominichamon do we follow the Google coding style here? (asking to learn, I kind of assumed it)

Reply:
yes we do.

Reply:
Ack  - @taoliq, that's why we pass by mutable pointer, because we'd only pass by reference if it were const, and that wouldn't be applicable here.

Reply:
the google style guide allows non-const references on parameters (since last year)

Reply:
Oh! Great! (also, finally!). Thanks!